File: Observer/Golang/AverageObserver.go
package main

import "fmt"

type AverageObserver struct {
	counter int
	resources [][]int
	averageCPU float32
	averageRAM float32
}

func (observer *AverageObserver) update(CPU int, RAM int)  {
	observer.resources = append(observer.resources, []int{CPU, RAM})
	observer.counter++
	tempCPU := 0
	tempRAM := 0

	for _, resource := range observer.resources {
		tempCPU += resource[0]
		tempRAM += resource[1]
	}

	observer.averageCPU = float32(tempCPU) / float32(observer.counter)
	observer.averageRAM = float32(tempRAM) / float32(observer.counter)

	fmt.Printf("Current Average CPU: %.2f\nCurrent Average RAM: %.2f\n", observer.averageCPU, observer.averageRAM)
}

File: Observer/Python 3/AverageObserver.py
from IObserver import IObserver

class AverageObserver(IObserver):
  def __init__(self):
    super().__init__()
    self.averageCPU = 0.0
    self.averageRAM = 0.0
    self.counter = 0
    self.resources = []
    
  def update(self, cpu, ram):
    self.resources.append((cpu, ram))
    tempCPU = 0.0
    tempRAM = 0.0
    self.counter += 1

    for resource in self.resources:
      tempCPU += resource[0]
      tempRAM += resource[1]
    
    self.averageCPU = tempCPU / self.counter
    self.averageRAM = tempRAM / self.counter

    print("Current Average CPU: {}\nCurrent Average RAM: {}".format(self.averageCPU, self.averageRAM))
